README.md

```diff
@@ -21,3 +21,141 @@ From a command line, execute the application to get full usage help.
+
+#### `output_file_name`
+Specify how the output files should be names here. The base ruleset contains the value `study_slide`. In this case, if my input slides are named: `john_smith_lung.svs` and `john_smith_pancreas.svs`, the redacted output images will be named `study_slide_1.svs` and `study_slide_2.svs`.
+
+### File Format Rules
+Redaction behavior is specified per file type. Currently pure `tiff` files and Aperio (`.svs`) files are supported. Each image type has its own groups of data that can be redacted. For example, Aperio images have `tiff` metadata, certain associated images, and additional metadata specified in the `ImageDescription` tag. `svs` rulesets take the following shape:
+
+```yaml
+svs:
+    associated_images:
+        ...
+    metadata:
+        ...
+    image_description:
+        ...
+```
+
+Each group is a dictionary whose keys represent a way to identify a specific piece of metadata or specific associated image, and whose values are dictionaries that define redaction behavior. Each entry (key-value pair) in the dictionary is a "rule." Take the following `associated_image` rule from the base ruleset
+
+```yaml
+svs:
+    ...
+    associated_images:
+        label:
+            action: replace
+            replace_with: blank_image
+    ...
+```
+
+This describes how `imagedephi` handles `label` images for Aperio files by default. Since label images frequently contain PHI, but are required by the Aperio (.svs) format, they are replaced with a black square of the same size.
+
+#### Image Rules
+
+Image rules take the following form:
+
+```yaml
+<image_key>:
+    action:
+```
+
+Where `image_key` identifies a particular associated image. For a catch-all rule, use the key `default`.
+
+Image rules can have the following actions:
+
+* `replace`: Replace an image with another. If specified, a value for `replace_with` must also be provided
+* `keep`: Does nothing. The associated image matching this key will be included in the output file
+* `delete`: The image will not be included in the output file
+
+For image rules, the only supported value of `replace_with` is `blank_image`.
+
+#### Metadata Rules
+
+Metadata rules take the following form:
+
+```yaml
+<metadata_key>:
+    action:
+```
+
+Where `metadata_key` identifies a piece of metadata. Possible values for this key depend on the type of metadata being redacted. For example, rules listed under
+
+```yaml
+tiff:
+    metadata:
+```
+have `metadata_keys` for particular tiff tags (e.g. `ImageDescription`, `ImageWidth`).
+
+Available actions for metadata rules are:
+
+* `delete`: the metadata will not appear in the output file
+* `keep`: the metadata will appear unchanged in the output file
+* `replace`: replace the metadata with a specified value. If this is the `action`, additional fields are required.
+* `check_type`: the metadata will appear in the output file if it is of the expected type. Requires additional fields
```

Comment:
Am I correct that the rule matches if the type matches but not if it doesn't?  That is, a match is `keep` but a failure to match is a rule miss, so you would add another rule to replace or delete those non-matching type entries?

Reply:
The initial implementation of the type check rules matches by tag/image description key. Behavior is branching based on the if the actual datatype lines up with the expected data type, but both are considered a "match."

I can see why this might be confusing. I think that if a type mismatch was a rule "miss," and another rule was required to fully redact some piece of metadata, it might be unintuitive. With the initial implementation, only one rule is needed to redact that piece of metadata. I'm open to changing how that type of rule works if the current implementation is lacking in some way.

Reply:
I think this means we need to clarify the text: `This will either keep the metadata if the type matches or delete the metadata if the type does not match`.  And, looking at the code, if we are using a type check of rational, should count be the number of rationals or the number of integers (that is, count of 1 would be two integers)?